### PR TITLE
Add channel normalization and pixel thresholding back

### LINKED ISSUE #980: Re-add channel normalization and pixel thresholding
**Is your feature request related to a problem? Please describe.**

Candace has created her own repo for the Pixie paper. It's been out for some time now, so now would be a good idea to add the channel and pixel normalization strategies back.

**Describe the solution you'd like**

#913 removed this, so just undo the changes.

**Additional context**

This will be helpful for testing Noah's TONIC cohort on the updated `ark-analysis` Docker image.

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Closes #980. Candace's paper has been out for some time now, so we can add these steps back.

**How did you implement your changes**

Revert to previous state with channel normalization and pixel thresholding.

### CODE REVIEW

**Review by srivarra — APPROVED** [review]

**Comment by cliu72** [comment]
Wait can we not merge this in for now?

Discussed with Noah, and we need to pin a new version for my paper that works before we merge this back in.

**Comment by cliu72** [comment]
Also, before we merge this back in, can we get these addressed (and then a new version pinned for my paper):

https://github.com/angelolab/ark-analysis/issues/992
https://github.com/angelolab/ark-analysis/issues/972
https://github.com/angelolab/ark-analysis/issues/969

**Comment by alex-l-kong** [comment]
@cliu72 Sri will handle #969, I'll take care of #972 and #992.

**Comment by alex-l-kong** [comment]
@cliu72 version 0.6.4 PR has been merged in with the update Pixie changes, I think this is ready to review.

**Review by camisowers — APPROVED** [review]
Looks right to me.

File: src/ark/phenotyping/pixie_preprocessing.py
Comment (cliu72) [inline src/ark/phenotyping/pixie_preprocessing.py:339]: I think `percentile` here should be a new variable called `channel_percentile_prenorm`. `channel_percentile_postnorm` is the percentile we are using to calculate the normalization values after pixel-level row normalization. This came up in Cami's pull request last time - we noticed that we were using 99% for the pre-row normalization, and 99.9% for the post-row normalization. I think I asked @ngreenwald awhile ago why he used 99 and not 99.9 and I think the answer was just that it looked better for his cohort. So I think solution for most flexibility is to have one more input variable (`channel_percnetile_prenorm`) to `create_pixel_matrix` and set it to 0.99 as a default.

File: src/ark/phenotyping/pixie_preprocessing.py
Comment (cliu72) [inline src/ark/phenotyping/pixie_preprocessing.py:262]: To be like super explicit, can we change this to `channel_norm_pre_rowsum.feather`? To match the corresponding `channel_norm_post_rowsum` file name.

Comment (cliu72) [reply src/ark/phenotyping/pixie_preprocessing.py:262]: Wait actually I just realized the default in the function call to `create_pixel_matrix` is `channel_norm_post_rowsum.feather` - can we change to `channel_norm_post_rownorm.feather`?

File: src/ark/phenotyping/pixie_preprocessing.py
Comment (cliu72) [inline src/ark/phenotyping/pixie_preprocessing.py:289]: Sorry to be annoying about all these filenames - can we change this to `channel_norm_post_rownorm_perfov.csv'?

**Review by cliu72 — CHANGES_REQUESTED** [review]
Looks good, just a few small requests

File: src/ark/phenotyping/pixie_preprocessing.py
Comment (alex-l-kong) [reply src/ark/phenotyping/pixie_preprocessing.py:262]: I'm a little confused: since we're updating this section: https://github.com/angelolab/ark-analysis/blob/3fe18bb7ee8e2b83402e67055e82533507deb5ba/src/ark/phenotyping/pixie_preprocessing.py#L331-L345, to use `channel_percentile_prenorm`, would `channel_norm_path` not contain the pre rowsum norm values? Or am I misunderstanding something?

Comment (cliu72) [reply src/ark/phenotyping/pixie_preprocessing.py:262]: Actually, I think it would be better to move `channel_norm.feather` (to be changed to `channel_norm_pre_rownorm.feather`) to be an input variable in the function call to `create_pixel_matrix`. Since this exists: `norm_vals_name='channel_norm_post_rowsum.feather'`, I think it would be better to make both input variables. So it would look something like this: 
```
def create_pixel_matrix(fovs, channels, base_dir, tiff_dir, seg_dir,
                        img_sub_folder="TIFs", seg_suffix='_whole_cell.tiff',
                        pixel_output_dir='pixel_output_dir',
                        data_dir='pixel_mat_data',
                        subset_dir='pixel_mat_subsetted',
                        is_mibitiff=False,
                        blur_factor=2, subset_proportion=0.1, seed=42
                        norm_vals_name_pre_rownorm='channel_norm_pre_rownorm.feather', #see this change
                        norm_vals_name_post_rownorm='channel_norm_post_rownorm.feather', #see this change
                        channel_percentile_pre_rownorm=0.99, #see this change
                        channel_percentile_post_rownorm = 0.999, #see this change
                        multiprocess=False, batch_size=5):
```

File: src/ark/phenotyping/pixie_preprocessing.py
Comment (alex-l-kong) [reply src/ark/phenotyping/pixie_preprocessing.py]: So just to clarify did you want to rename the actual parameter argument `channel_percentile_prenorm` or just assign `percentile=channel_percentile_prenorm` instead?

Comment (cliu72) [reply src/ark/phenotyping/pixie_preprocessing.py]: Assigning `percentile=channel_percentile_prenorm` is fine

File: src/ark/phenotyping/pixie_preprocessing.py
Comment (cliu72) [inline src/ark/phenotyping/pixie_preprocessing.py:231]: For clarity, can we change this to "The name of the file to store the pre pixel-normalization channel norm values"

File: src/ark/phenotyping/pixie_preprocessing.py
Comment (cliu72) [inline src/ark/phenotyping/pixie_preprocessing.py:233]: Same here. For clarity, can we change this to "The name of the file to store the post-pixel normalization channel norm values"

File: src/ark/phenotyping/pixie_preprocessing.py
Comment (cliu72) [inline src/ark/phenotyping/pixie_preprocessing.py:237]: Same here. For clarify, can we change this to "Percentile used to normalize channels before pixel normalization"

File: src/ark/phenotyping/pixie_preprocessing.py
Comment (cliu72) [inline src/ark/phenotyping/pixie_preprocessing.py:239]: Same here. For clarify, can we change this to "Percentile used to normalize channels after pixel normalization"

**Review by cliu72 — COMMENTED** [review]
Just a few tiny requests on text change for variable descriptions

**Review by cliu72 — APPROVED** [review]
Looks good!

**Pull request merged**
